### PR TITLE
Feature/manifest api

### LINKED ISSUE #146: Election Manifest API
On Election Setup Service
- Get existing manifest from data store based on election_id
- Upload AND validate manifest and return success or failure
- The validation should use JSON schema or similar validation on the top end and try to parse the Manifest into electionguard-python

### PR DESCRIPTION
**Pull request by AddressXception**
### Issue
*Link your PR to an issue*

Fixes #146

### Description
Adds submit manifest and get manifest endpoints

### Testing
1. using postman
2. put manifest
3. get manifest

Expected: the manifest you put is returned

### CODE REVIEW

File: app/api/v1/mediator/manifest.py [hunk 1]
```diff
@@ -1,53 +1,142 @@
...
+    ValidateManifestRequest,
+    BaseValidationResponse,
+    ResponseStatus,
+)
 from ..tags import MANIFEST
 
 router = APIRouter()
 
+# TODO: multi-tenancy
+CLIENT_ID = "electionguard-default-client-id"
```
Comment (AddressXception) [inline app/api/v1/mediator/manifest.py:25]: i figure this stub is ok for now.  we'll need a way to handle multi-tenancy at some point but I'd be inclined to make an issue for this and handle it later once I hook everything up in the election context PR

File: app/api/v1/mediator/manifest.py [hunk 2]
```diff
@@ -1,53 +1,142 @@
...
+            )
+    except Exception as error:
+        print(sys.exc_info())
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="get manifest failed",
+        ) from error
+
+
+@router.put("", tags=[MANIFEST], status_code=status.HTTP_204_NO_CONTENT)
```
Comment (AddressXception) [inline app/api/v1/mediator/manifest.py:57]: whoops, this should be a 202

File: app/api/v1/mediator/manifest.py [hunk 3]
```diff
@@ -1,53 +1,142 @@
...
+            query_result = repository.get({"manifest_hash": crypto_hash.to_hex()})
+            if not query_result:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail=f"Could not find manifest {manifest_hash}",
+                )
+
+            return ManifestQueryResponse(
+                status=ResponseStatus.SUCCESS,
+                manifest=query_result["manifest"],
```
Comment (AddressXception) [inline app/api/v1/mediator/manifest.py:47]: i'm not thrilled about sidestepping the typing but I didnt want to go through the boilerplate of generalizing the repository interface just yet so i think this is fine for now

File: app/api/v1/mediator/manifest.py (hunk 1 above)
Comment (AddressXception) [reply app/api/v1/mediator/manifest.py:25]: my general theory here is that when we have authorization then all of the election state stuff is tied to a tenant, and then all of the election-specific stuff is tied to that election state, so you can't query the ballot stuff if you can't resolve the election state

File: app/api/v1/mediator/manifest.py [hunk 4]
```diff
@@ -1,53 +1,142 @@
...
+    _, response = _validate_manifest(request, schema)
+    return response
+
 
+def _deserialize_manifest(data: object) -> Optional[Manifest]:
+    try:
+        return Manifest.from_json_object(data)
+    except Exception:  # pylint: disable=broad-except
+        # TODO: some sort of information why it failed
+        return None
```
Comment (AddressXception) [inline app/api/v1/mediator/manifest.py:108]: we are going to need to add exceptions into the core python repo for this part but it can wait until later

**Review by SteveMaier-IRT — APPROVED** [review]

**Pull request merged**